components/brave_new_tab_ui/containers/newTab/index.tsx

```diff
@@ -29,23 +30,50 @@ interface Props {
   saveShowClock: (value: boolean) => void
   saveShowTopSites: (value: boolean) => void
   saveShowStats: (value: boolean) => void
+  saveShowRewards: (value: boolean) => void
 }
 
 interface State {
+  onlyAnonWallet: boolean
   showSettingsMenu: boolean
   backgroundHasLoaded: boolean
 }
 
 class NewTabPage extends React.Component<Props, State> {
   state = {
+    onlyAnonWallet: false,
     showSettingsMenu: false,
     backgroundHasLoaded: false
   }
 
   componentDidMount () {
+    const { walletCreated } = this.props.newTabData.rewardsState
+
     // if a notification is open at component mounting time, close it
     this.props.actions.onHideSiteRemovalNotification()
     this.trackCachedImage()
+
+    chrome.braveRewards.getAdsEnabled((enabled: boolean) => {
+      this.props.actions.onAdsEnabled(enabled)
+    })
+    chrome.braveRewards.getRewardsMainEnabled((enabled: boolean) => {
+      this.props.actions.onEnabledMain(enabled)
+    })
+
+    if (!walletCreated) {
+      return
+    }
+
+    chrome.braveRewards.onlyAnonWallet((only: boolean) => {
+      this.setState({
+        onlyAnonWallet: !!only
+      })
+    })
+
+    this.refreshActions()
+    window.setInterval(() => {
+      this.refreshActions()
+    }, 30000)
```

Comment:
Putting this in componentDidMount seems like an anti-pattern:
1) there's no reason this cannot be done separate to the rendering outside of the component in the root of this module, or in the root page module, or preferably in an init function.
2) Calling setState very quickly afterwards will cause another render cycle.
3) There are 6 independent redux actions dispatched immediately after, causing another 6 render cycles.
4) These events are also firing when the rewards widget is turned off

This, combined with the other refreshActions comment below, seems like a good candidate to move to a simple `getRewardsInitData` and `getRewardsUpdateData` function, action and reducer combo. They could also listen to the NTP prefs change event and only dispatch the rewards-related actions if the rewards widget is turned on, or if you want to keep them dispatching redux actions all the time then the reducer could manipulate state only when the rewards widget pref is on.

Reply:
implemented per discussion in slack